cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/ReconcilerUtils.java

```diff
@@ -381,4 +381,24 @@ public static boolean nodePoolsEnabled(Kafka kafka) {
     public static boolean kraftEnabled(Kafka kafka) {
         return KafkaCluster.ENABLED_VALUE_STRIMZI_IO_KRAFT.equals(Annotations.stringAnnotation(kafka, Annotations.ANNO_STRIMZI_IO_KRAFT, "disabled").toLowerCase(Locale.ENGLISH));
     }
+
+    /**
+     * Creates a hash from Secret's data.
+     * 
+     * @param secret Secret with data.
+     * @return Hash of all secret values.
+     */
+    public static String hashSecretContent(Secret secret) {
+        if (secret == null) {
+            throw new RuntimeException("Secret not found");
+        }
+        if (secret.getData() == null || secret.getData().isEmpty()) {
+            throw new RuntimeException("Empty secret");
+        }
+        StringBuilder sb = new StringBuilder();
+        for (String key : secret.getData().keySet()) {
+            sb.append(secret.getData().get(key));
+        }
+        return Util.hashStub(sb.toString());
+    }
```

Comment:
A nit, but if you add some spacing, it would be easier to read:

```suggestion
    public static String hashSecretContent(Secret secret) {
        if (secret == null) {
            throw new RuntimeException("Secret not found");
        }
        
        if (secret.getData() == null || secret.getData().isEmpty()) {
            throw new RuntimeException("Empty secret");
        }
        
        StringBuilder sb = new StringBuilder();
        for (String key : secret.getData().keySet()) {
            sb.append(secret.getData().get(key));
        }
        
        return Util.hashStub(sb.toString());
    }
```